### ISSUE TITLE
'boost/preprocessor/control/expr_iif.hpp' file not found when run "npx expo run:ios"

### ISSUE THREAD

**Issue by heryTz** [expo/expo#26942]
### Summary

The build doesn't work, it seems to me that there are missing files in some Pod dependencies.

### What platform(s) does this occur on?

iOS

### SDK Version

50.0.6

### Environment

expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 13.6.4
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.11.0 - /usr/local/bin/node
      Yarn: 1.22.21 - /usr/local/bin/yarn
      npm: 10.2.4 - /usr/local/bin/npm
      Watchman: 2024.01.22.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.15.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, visionOS 1.0, watchOS 10.2
    IDEs:
      Android Studio: 2023.1 AI-231.9392.1.2311.11330709
      Xcode: 15.2/15C500b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~50.0.6 => 50.0.6 
      react: 18.2.0 => 18.2.0 
      react-native: 0.73.3 => 0.73.3 
    npmGlobalPackages:
      eas-cli: 7.1.2
    Expo Workflow: bare

### Minimal reproducible example

[https://github.com/heryTz/expo-ios-build-error](https://github.com/heryTz/expo-ios-build-error)

**Comment by hery-tech** [comment]
I don't know if it's related to this, but "expo-image" also gives a similar error during compilation.

```bash
❌  (ios/Pods/libaom/aom/aom_dsp/psnrhvs.c:24:10)

  22 | 
  23 | #include "aom_dsp/psnr.h"
> 24 | #include "aom_dsp/ssim.h"
     |          ^ 'aom_dsp/ssim.h' file not found
  25 | #include "aom_ports/system_state.h"
  26 | 
  27 | static void od_bin_fdct8x8(tran_low_t *y, int ystride, const int16_t *x,
```

**Comment by heryTz** [comment]
I don't know if it's related to this, but "expo-image" also gives a similar error during compilation.

```bash
❌  (ios/Pods/libaom/aom/aom_dsp/psnrhvs.c:24:10)

  22 | 
  23 | #include "aom_dsp/psnr.h"
> 24 | #include "aom_dsp/ssim.h"
     |          ^ 'aom_dsp/ssim.h' file not found
  25 | #include "aom_ports/system_state.h"
  26 | 
  27 | static void od_bin_fdct8x8(tran_low_t *y, int ystride, const int16_t *x,
```

**Comment by rlesniak** [comment]
Also started to happen today. Even raw app from `npx create-expo-app` has error `/expr_iif.hpp' file not found`

**Comment by hirbod** [comment]
For me, it randomly says "Pod xyz not found". After 10x trying `npx pod-install`, the issue disappeared but now it tells me `'aom_dsp/ssim.h' file not found`. Something is strange. Also expo-image

**Comment by hirbod** [comment]
I confirmed the issue was caused by `brew` and Cocoapods 1.15.1. Downgrading is not straightforward when the formulas are not available anymore. I removed Cocoapods from brew and used `asdf` to install the current `ruby` version. You can use any package manager you like, just make sure that you don't rum `gem` as sudo. 

Here are the steps to fix it:

```bash
brew uninstall cocoapods
brew autoremove
asdf plugin add ruby
asdf install ruby latest (3.3.0)
asdf install ruby 3.3.0
restart terminal
gem install cocoapods -v 1.14.3
gem update --system 3.5.6
npx expo prebuild --clean
npx expo run:ios  
```

**Comment by byCedric** [comment]
Hi all! It seems that Cocoapods just reverted a couple of offending PRs and released `1.15.2` ([release](https://github.com/CocoaPods/CocoaPods/releases/tag/1.15.2)). This should resolve the issue mentioned above.

**Comment by huseyintopuz** [comment]
Hi !. is this answer is valid for native cli
